components/adv-post-form.js

```diff
@@ -26,9 +28,116 @@ const FormStatus = {
+      <li>The highest boost across all territories over the last 30 days is pinned to the top of the homepage</li>
+      <li>The minimum boost is {numWithUnits(BOOST_MIN, { abbreviate: false })}</li>
+      <li>Each {numWithUnits(BOOST_MULT, { abbreviate: false })} of boost is equivalent to a zap-vote from a maximally trusted stacker
+        <ul>
+          <li>e.g. {numWithUnits(BOOST_MULT * 5, { abbreviate: false })} is like five zap-votes from a maximally trusted stacker</li>
+        </ul>
+      </li>
+      <li>The decay of boost "votes" increases at 1.25x the rate of organic votes
+        <ul>
+          <li>i.e. boost votes fall out of ranking faster</li>
+        </ul>
+      </li>
+      <li>boost can take a few minutes to show higher ranking in feed</li>
+      <li>100% of boost goes to the territory founder and top stackers as rewards</li>
+    </ol>
+  )
+}
+
+export function BoostInput ({ onChange, ...props }) {
+  const feeButton = useFeeButton()
+  let merge
+  if (feeButton) {
+    ({ merge } = feeButton)
+  }
+  return (
+    <Input
+      label={
+        <div className='d-flex align-items-center'>boost
+          <Info>
+            <BoostHelp />
+          </Info>
+        </div>
+    }
+      name='boost'
+      onChange={(_, e) => {
+        merge?.({
+          boost: {
+            term: `+ ${e.target.value}`,
+            label: 'boost',
+            op: '+',
+            modifier: cost => cost + Number(e.target.value)
+          }
+        })
+        onChange && onChange(_, e)
+      }}
+      hint={<span className='text-muted'>ranks posts higher temporarily based on the amount</span>}
+      append={<InputGroup.Text className='text-monospace'>sats</InputGroup.Text>}
+      {...props}
+    />
+  )
+}
+
+// act means we are adding to existing boost
+export function BoostItemInput ({ item, sub, act = false, ...props }) {
+  const [boost, setBoost] = useState(Number(item?.boost) + (act ? BOOST_MULT : 0))
+
+  const [getBoostPosition, { data }] = useLazyQuery(gql`
+    query BoostPosition($id: ID, $boost: Int) {
+      boostPosition(sub: "${item?.subName || sub?.name}", id: $id, boost: $boost) {
+        home
+        sub
+      }
+    }`,
+  { fetchPolicy: 'cache-and-network' })
+
+  const getPositionDebounce = useDebounceCallback((...args) => getBoostPosition(...args), 1000, [getBoostPosition])
+
+  useEffect(() => {
+    if (boost) {
+      getPositionDebounce({ variables: { boost: Number(boost), id: item?.id } })
+    }
+  }, [boost, item?.id])
+
+  const boostMessage = useMemo(() => {
+    if (data?.boostPosition?.home || data?.boostPosition?.sub) {
+      const boostPinning = []
+      if (data?.boostPosition?.home) {
+        boostPinning.push('homepage')
+      }
+      if (data?.boostPosition?.sub) {
+        boostPinning.push(`~${item?.subName || sub?.name}`)
+      }
+      return `pins to the top of ${boostPinning.join(' and ')}`
```

Comment:
This shows `pins to the top of ~undefined` for comments because `boostPosition.sub` is true but `item.subName` and `sub.name` are undefined.

I think comment boosts are cool but this scoring shouldn't run for comments or we should include a item score.

Reply:
Fixed on master